org.eclipse.xtext.builder.standalone/src/org/eclipse/xtext/builder/standalone/StandaloneBuilder.java

```diff
@@ -344,6 +344,22 @@ public boolean launch() {
 		}
 	}
 
+	/**
+	 * Allows for relative source folders are properly generated in the
+	 * trace files, instead of absolute ones.
+	 * 
+	 * See https://github.com/eclipse/xtext/issues/2957
+	 * 
+	 * @since 2.35
+	 */
+	protected void configureWorkspace(ResourceSet resourceSet) {
+		FileProjectConfig projectConfig = new FileProjectConfig(new File(baseDir), baseDir);
```

Comment:
https://github.com/eclipse/xtext/blob/main/org.eclipse.xtend.core/src/org/eclipse/xtend/core/compiler/batch/XtendBatchCompiler.java#L412 doesn't check for that, but maybe it's better to in this class, otherwise `install` would thrown an IllegalStateException.

btw, I also have to rebase to avoid the merge conflict